src/librustc/dep_graph/graph.rs

```diff
@@ -231,7 +242,16 @@ impl DepGraph {
     pub fn read(&self, v: DepNode) {
         if let Some(ref data) = self.data {
             data.edges.borrow_mut().read(v);
-            data.current.borrow_mut().read(v);
+
+            let mut current = data.current.borrow_mut();
+            debug_assert!(current.node_to_node_index.contains_key(&v),
```

Comment:
Why have *both* the debug assert *and* the `bug!`?

Also, what is going on here =)